Gems/ROS2/Code/Source/RobotImporter/xacro/XacroUtils.cpp

```diff
@@ -0,0 +1,142 @@
+#include <QProcess>
+#include <QString>
+
+namespace ROS2::Utils::xacro
+{
+
+    ExecutionOutcome ParseXacro(const AZStd::string& filename, const Params& params)
+    {
+        ExecutionOutcome outcome;
+        // test if xacro exists
+        AZStd::string xacroPath = WhichXacro();
+
+        AZ_Warning("ParseXacro", !xacroPath.empty(), "There is xacro executable in your path");
+        if (xacroPath.empty())
+        {
+            return ExecutionOutcome{};
+        }
+        AZ_Printf("ParseXacro", "xacro executable : %s", xacroPath.c_str());
+        AZ_Printf("ParseXacro", "Convert xacro file : %s", filename.c_str());
+        const QString program = QString::fromUtf8(xacroPath.data(), xacroPath.size());
+        QStringList arguments;
+        arguments << QString::fromUtf8(filename.data(), int(filename.size()));
+
+        for (const auto& p : params)
+        {
+            QString name = QString::fromUtf8(p.first.data());
+            QString value = QString::fromUtf8(p.second.data());
+            arguments << name + ":=" + value;
+        }
+
+        QProcess myProcess;
+
+        QString called = program;
+
+        for (const auto& p : arguments)
+        {
+            called += " " + p;
+        }
+
+        outcome.m_called = AZStd::string(called.toUtf8().constData());
+        AZ_Printf("ParseXacro", "calling file : %s", outcome.m_called.c_str());
+
+        myProcess.start(program, arguments);
+
+        if (myProcess.waitForFinished() && myProcess.exitStatus() == QProcess::NormalExit && myProcess.exitCode() == 0)
+        {
+            AZ_Printf("ParseXacro", "xacro finished with success");
+            auto output = myProcess.readAllStandardOutput();
+            outcome.m_urdfHandle = UrdfParser::Parse(output.data());
+        }
+        else
+        {
+            AZ_Printf("ParseXacro", "xacro finished with code %d", myProcess.exitCode());
+            auto std = myProcess.readAllStandardOutput();
+            auto cerr = myProcess.readAllStandardError();
+            outcome.m_logStandardOutput = AZStd::string(std.data(), std.size());
+            outcome.m_logErrorOutput = AZStd::string(cerr.data(), cerr.size());
+        }
+        outcome.m_returnCode = myProcess.exitCode();
+        return outcome;
+    }
+
+    AZStd::string WhichXacro()
+    {
+        const QString program{ "which" };
+        QStringList arguments;
+        arguments << kXacroExecutable;
+        QProcess myProcess;
+        myProcess.start(program, arguments);
+        if (myProcess.waitForFinished() && myProcess.exitStatus() == QProcess::NormalExit && myProcess.exitCode() == 0)
+        {
+            auto output = myProcess.readAllStandardOutput().trimmed();
+            return AZStd::string{ output.data(), static_cast<AZStd::string::size_type>(output.size()) };
+        }
+        return AZStd::string{};
+    }
+
+    AZStd::unordered_map<AZStd::string, AZStd::string> GetParameterFromXacroData(AZStd::vector<char> data)
```

Comment:
It would be better to accept an `AZStd::span<char>` here and that would allow the any type of contiguous structure of `char` to be supplied to this method(`AZStd::vector<char>`, `AZStd::array<char, N>`, `AZStd::string`, `AZStd::fixed_string<N>`, `AZStd::string_view`, `AZStd::fixed_vector<char, N>`, etc...)

Here is the API for https://en.cppreference.com/w/cpp/container/span